### ISSUE TITLE
introducing `ListLoggingT`?

### ISSUE THREAD

**Issue by marcosh** [ocharles/logging-effect#43]
Oftentimes if find myself defining datatype which I want to log a sum types, like

```haskell
data Warning = FooWarning | BarWarning | BazWarning
```

This is fine as long as I use `MonadLog` instances as `LoggingT Warning` which process messages one by one.

When in tests I want to use `PureLoggingT`, I need to have a monoid instance and I'm forced to change the message type to `[Warning]`. But this kind of pollutes the production code where I end up having `[Warning]`, too.

This would be solved by having something like `ListLoggingT` which uses the free monoid for the message type, something like

```haskell
newtype ListLoggingT log m a = MkListLoggingT (StateT [log] m a)

runListLoggingT :: ListLoggingT log m a -> m (a, [log])
```

Is there a better solution? Or a way to just use `PureLoggingT` composing it with something else?

Or, if this makes sense, would you be interested for a PR adding `ListLoggingT` to the project?

**Comment by ocharles** [comment]
Hi @marcosh!

Unfortunately I'm no longer using it maintaining this library so there's not much I can do to help. I think what I've done in the past here is use a `mapLogMessage` to map log messages into a type like `Seq`, then you can use pure logging.

Also, if you're using this library and want to take it over or maintain it, just let me know - I'll be happy to add you as a contributor!

**Comment by marcosh** [comment]
Thanks for the hint with `Seq`. I tried it but in the end it felt a bit too complex (I had too use `mapLoggingT` and then `mapLogMessage`) and in the end I decided that `ListLoggingT` felt simpler.

> Also, if you're using this library and want to take it over or maintain it, just let me know - I'll be happy to add you as a contributor!

Thanks for the offer. We just started to use the library and, for the moment, I don't feel it would be a good move to take over or maintain the library. If in the future we'll need more changes/updates in the library, I'll ping you again
